src/main/webapp/app/shared/metis/posting-reactions-bar/posting-reactions-bar.directive.ts

```diff
@@ -121,6 +134,96 @@ export abstract class PostingsReactionsBarDirective<T extends Posting> implement
         this.metisService.deletePost(this.posting);
     }
 
+    openForwardMessageView(post: Posting, isAnswer: boolean): void {
+        if (!this.course()?.id) {
+            return;
+        }
+        this.channels = [];
+        this.users = [];
+
+        this.conversationService
+            .getConversationsOfUser(this.course()!.id!)
+            .pipe(map((response) => response.body || []))
+            .subscribe({
+                next: (conversations) => {
+                    conversations.forEach((conversation) => {
+                        if (conversation.type === ConversationType.CHANNEL) {
+                            this.channels.push(conversation as ChannelDTO);
+                        }
+                    });
+
+                    const modalRef = this.modalService.open(ForwardMessageDialogComponent, {
+                        size: 'lg',
+                        backdrop: 'static',
+                    });
+
+                    modalRef.componentInstance.users.set([]);
+                    modalRef.componentInstance.channels.set(this.channels);
+                    modalRef.componentInstance.postToForward.set(post);
+                    modalRef.componentInstance.courseId.set(this.course()?.id);
+
+                    modalRef.result.then(
+                        async (selection: { channels: Conversation[]; users: UserPublicInfoDTO[]; messageContent: string }) => {
+                            if (selection) {
```

Comment:
Maybe its also possible here to remove some of the nesting by inverting ifs.

Reply:
I chose not to refactor this for now. Although these ifs might not be necessary, I think the code seems more understandable in its current form